ambry-router/src/main/java/com.github.ambry.router/AdaptiveOperationTracker.java

```diff
@@ -219,7 +224,8 @@ private Counter getWholeDcPastDueCounter(RouterOperation routerOperation) {
 
     @Override
     public boolean hasNext() {
-      return replicaIterator.hasNext() && (inflightCount < parallelism || isOldestRequestPastDue());
+      return replicaIterator.hasNext() && (inflightCount < parallelism || (
```

Comment:
It might be more readable to break this into multiple if statements:
```java
      if (replicaIterator.hasNext()) {
        if (inflightCount < parallelism) {
          return true;
        }
        if (inflightCount < routerConfig.routerAdaptiveTrackerMaxInflightRequests && isOldestRequestPastDue()) {
          return true;
        }
      }
      return false;
```

Up to your personal preference, though

Reply:
True, your code is more readable. I will take your advice.